torchqdynamics/utils/td_tensor.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import get_args
+
+import torch
+from torch import Tensor
+
+from .tensor_types import OperatorLike, TDOperatorLike, cdtype, rdtype, to_tensor
+
+
+def to_td_tensor(
+    x: TDOperatorLike,
+    dtype: torch.dtype | None = None,
+    device: torch.device | None = None,
+    is_complex: bool = False,
+) -> TDTensor:
+    """Convert a `TDOperatorLike` object to a `TDTensor` object."""
+    if isinstance(x, get_args(OperatorLike)):
+        # convert to tensor
+        x = to_tensor(x, dtype=dtype, device=device, is_complex=is_complex)
+
+        return ConstantTDTensor(x)
+    elif callable(x):
+        # get default dtype and device
+        if dtype is None:
+            if is_complex:
+                dtype = cdtype(dtype)
+            else:
+                dtype = rdtype(dtype)
+        if device is None:
+            device = get_default_device()
+
+        # check callable
+        x0 = check_callable(x, expected_dtype=dtype, expected_device=device)
+
+        return CallableTDTensor(x, shape=x0.shape, dtype=dtype, device=device)
+
+
+def get_default_device() -> torch.device:
+    """Get the default device."""
+    return torch.ones(1).device
+
+
+def check_callable(
+    f: callable,
+    *,
```

Comment:
What is the purpose of `*` here? I understand what it does in python, but not what it brings

Reply:
True, I just like the feature so I tend to put it everywhere haha. But I guess it doesn't make sense here.